spec/system/visitor_views_post_spec.rb

```diff
@@ -23,4 +23,25 @@
       expect(page).to have_content post.article.content
     end
   end
+
+  context "by viewing post on category show page" do
+    it "shows truncated post" do
+      category = create(:category)
+      content = "It's important to me that you're happy. Maybe we got a few little happy bushes here, just covered with snow. It's all a game of angles. Anytime you learn something your time and energy are not wasted. You can do anything your heart can imagine. Do an almighty painting with us. We tell people sometimes: we're like drug dealers, come into town and get everybody absolutely addicted to painting. It doesn't take much to get you addicted. I guess that would be considered a UFO. A big cotton ball in the sky. Automatically, all of these beautiful, beautiful things will happen."
+      article = create(:article, content: content)
+      tag = create(:tag)
+
+      post = create(:post,
+        categories: [category],
+        article: article,
+        tags: [tag])
+
+      visit category_path(category)
+
+      expect(page).to have_content post.title
+      expect(page).to have_content post.published_at
+      expect(page).to have_content(post.article.content.truncate(200, separator: " ") { link_to t("posts.truncated.keep_reading"), post_path(post) })
```

Comment:
Is it worth having another expectation to ensure the page is *only* showing the truncated content?

```ruby
expect(page).not_to have_content(post.article.content)
```

Reply:
Ah, yes!  Good call!